glib/src/object.rs

```diff
@@ -1095,6 +1095,14 @@ impl Object {
         unsafe { Object::new_internal(type_, &params) }
     }
 
+    pub fn new_subclass<T: IsA<Object>>(
```

Comment:
I'm not entirely happen with the name. It's not only for "subclasses" but it's another way of creating any `GObject`.

Maybe this can become the new `new()` and the old `new()` could become `(new_)for_type()`? I expect the one with the `T: IsA<Object>` to be the most commonly used one.

Reply:
Isn't any type that `IsA<Object>` by definition a subclass of `Object`? But I'm not particularly enamoured with the name either.

> Maybe this can become the new new() and the old new() could become (new_)for_type()? I expect the one with the T: IsA<Object> to be the most commonly used one.

That could work.

Reply:
Then there's the existing `new_generic()` function. Not sure how `Value` is more "generic" than `&dyn ToValue`...

Reply:
> Isn't any type that IsA<Object> by definition a subclass of Object? But I'm not particularly enamoured with the name either.

Yes but the way you called it, it sounds like it's only for creating subclasses defined in Rust. `ObjectSubclass` impls :)

> Then there's the existing new_generic() function. Not sure how Value is more "generic" than &dyn ToValue...

If you have better suggestions... I don't like that one either :)

Reply:
Should maybe call the "generic" functions something with `value` :)

Reply:
@ids1024 What do you think? Let's do that as part of a different PR though.

Reply:
I have trouble judging it because I'm not sure exactly why `new_generic` would be used (it's not used in any test or example in this repo).

`Value` implements `ToValue`. Don't know if it's useful that it takes an owned `Value` rather than an `&` reference...

If it's useful, I guess `new()` and `with_type` could both have alternatives like that, but that's kind of annoying, hard to name, and I don't know exactly why it would be used.

Reply:
> Value implements ToValue. Don't know if it's useful that it takes an owned Value rather than an & reference...

That gives you a `Value` that contains a `Value`. It's not a no-op but wraps it one more time.

> I don't know exactly why it would be used.

If you have a `Value` you need that API. For example if you just want to proxy a value you got from `get_property()` to `set_property_generic()`.